### ISSUE TITLE
Bleadperl breaks Exception::SEH

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#11229]
Migrated from [rt.perl.org#87702](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=87702) (status was 'rejected')

Searchable as RT87702$

**Comment by p5pRT** [comment]
### From @cpansprout

See the CPAN ticket&#8203;: https://rt.cpan.org/Ticket/Display.html?id=66984

cc \-c 
\-I/home/cpansand/var/ctps/2pre5140/install/perl\-167630b6ab7e291cbd4f89943a3aec8d6a1ecbfc/lib/site\_perl/5\.13\.11/i386\-freebsd/B/Hooks/OP/Check/Install
\-I/home/cpansand/var/ctps/2pre5140/install/perl\-167630b6ab7e291cbd4f89943a3aec8d6a1ecbfc/lib/site\_perl/5\.13\.11/i386\-freebsd/B/Hooks/OP/PPAddr/Install
 \-DHAS\_FPSETMASK \-DHAS\_FLOATINGPOINT\_H \-fno\-strict\-aliasing \-pipe
\-fstack\-protector \-I/usr/local/include \-O    \-DVERSION=\\"0\.0201\\" 
\-DXS\_VERSION=\\"0\.0201\\" \-DPIC \-fPIC
"\-I/home/cpansand/var/ctps/2pre5140/install/perl\-167630b6ab7e291cbd4f89943a3aec8d6a1ecbfc/lib/5\.13\.11/i386\-freebsd/CORE"
&nbsp;  SEH\.c
SEH\.xs&#8203;: In function 'check\_return'&#8203;:
SEH\.xs&#8203;:36&#8203;: error&#8203;: 'Perl\_pp\_return' undeclared \(first use in this function\)
SEH\.xs&#8203;:36&#8203;: error&#8203;: \(Each undeclared identifier is reported only once
SEH\.xs&#8203;:36&#8203;: error&#8203;: for each function it appears in\.\)
\*\*\* Error code 1

I think this is because this commit moves the declaration of Perl\_pp\_\* functions into pp\_proto\.h&#8203;:

http&#8203;://perl5\.git\.perl\.org/perl\.git/commitdiff/897d39893

Why is the \#include in proto\.h inside an \#ifdef PERL\_CORE?


---
Flags&#8203;:
&nbsp;    category=core
&nbsp;    severity=high


---
Site configuration information for perl 5\.13\.11&#8203;:

Configured by sprout at Thu Mar 24 14&#8203;:02&#8203;:46 PDT 2011\.

Summary of my perl5 \(revision 5 version 13 subversion 11\) configuration&#8203;:
&nbsp;  Snapshot of&#8203;: cc13eef1fd2ed32ab3ff15ac5caa93d2badfd986
&nbsp;  Platform&#8203;:
&nbsp;    osname=darwin\, osvers=10\.5\.0\, archname=darwin\-thread\-multi\-2level
&nbsp;    uname='darwin pint\.local 10\.5\.0 darwin kernel version 10\.5\.0&#8203;: fri nov 5 23&#8203;:20&#8203;:39 pdt 2010; root&#8203;:xnu\-1504\.9\.17~1release\_i386 i386 '
&nbsp;    config\_args='\-Dusedevel \-de \-Duseithreads \-Doptimize=\-g'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=define\, usemultiplicity=define
&nbsp;    useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
&nbsp;    use64bitint=undef\, use64bitall=undef\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, ccflags ='\-fno\-common \-DPERL\_DARWIN \-no\-cpp\-precomp \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include'\,
&nbsp;    optimize='\-g'\,
&nbsp;    cppflags='\-no\-cpp\-precomp \-fno\-common \-DPERL\_DARWIN \-no\-cpp\-precomp \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='4\.2\.1 \(Apple Inc\. build 5664\)'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='env MACOSX\_DEPLOYMENT\_TARGET=10\.3 cc'\, ldflags =' \-fstack\-protector \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /usr/lib
&nbsp;    libs=\-ldbm \-ldl \-lm \-lutil \-lc
&nbsp;    perllibs=\-ldl \-lm \-lutil \-lc
&nbsp;    libc=/usr/lib/libc\.dylib\, so=dylib\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version=''
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=bundle\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;    cccdlflags=' '\, lddlflags=' \-bundle \-undefined dynamic\_lookup \-L/usr/local/lib \-fstack\-protector'

Locally applied patches&#8203;:
&nbsp;    



---
@&#8203;INC for perl 5\.13\.11&#8203;:
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.13\.11/darwin\-thread\-multi\-2level
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.13\.11
&nbsp;    /usr/local/lib/perl5/5\.13\.11/darwin\-thread\-multi\-2level
&nbsp;    /usr/local/lib/perl5/5\.13\.11
&nbsp;    /usr/local/lib/perl5/site\_perl
&nbsp;    \.



---
Environment for perl 5\.13\.11&#8203;:
&nbsp;    DYLD\_LIBRARY\_PATH \(unset\)
&nbsp;    HOME=/Users/sprout
&nbsp;    LANG=en\_US\.UTF\-8
&nbsp;    LANGUAGE \(unset\)
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)
&nbsp;    PATH=/usr/bin&#8203;:/bin&#8203;:/usr/sbin&#8203;:/sbin&#8203;:/usr/local/bin&#8203;:/usr/X11/bin&#8203;:/usr/local/bin
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    SHELL=/bin/bash

**Comment by p5pRT** [comment]
### From @nwc10

On Sun\, Apr 03\, 2011 at 02&#8203;:45&#8203;:52PM \-0700\, Father Chrysostomos wrote&#8203;:
> \# New Ticket Created by  Father Chrysostomos 
> \# Please include the string&#8203;:  \[perl \#87702\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=87702 >
> 
> 
> See the CPAN ticket&#8203;: https://rt.cpan.org/Ticket/Display.html?id=66984
> 
> cc \-c 
> \-I/home/cpansand/var/ctps/2pre5140/install/perl\-167630b6ab7e291cbd4f89943a3aec8d6a1ecbfc/lib/site\_perl/5\.13\.11/i386\-freebsd/B/Hooks/OP/Check/Install
> \-I/home/cpansand/var/ctps/2pre5140/install/perl\-167630b6ab7e291cbd4f89943a3aec8d6a1ecbfc/lib/site\_perl/5\.13\.11/i386\-freebsd/B/Hooks/OP/PPAddr/Install
>  \-DHAS\_FPSETMASK \-DHAS\_FLOATINGPOINT\_H \-fno\-strict\-aliasing \-pipe
> \-fstack\-protector \-I/usr/local/include \-O    \-DVERSION=\\"0\.0201\\" 
> \-DXS\_VERSION=\\"0\.0201\\" \-DPIC \-fPIC
> "\-I/home/cpansand/var/ctps/2pre5140/install/perl\-167630b6ab7e291cbd4f89943a3aec8d6a1ecbfc/lib/5\.13\.11/i386\-freebsd/CORE"
>   SEH\.c
> SEH\.xs&#8203;: In function 'check\_return'&#8203;:
> SEH\.xs&#8203;:36&#8203;: error&#8203;: 'Perl\_pp\_return' undeclared \(first use in this function\)
> SEH\.xs&#8203;:36&#8203;: error&#8203;: \(Each undeclared identifier is reported only once
> SEH\.xs&#8203;:36&#8203;: error&#8203;: for each function it appears in\.\)
> \*\*\* Error code 1
> 
> I think this is because this commit moves the declaration of Perl\_pp\_\* functions into pp\_proto\.h&#8203;:
> 
> http&#8203;://perl5\.git\.perl\.org/perl\.git/commitdiff/897d39893
> 
> Why is the \#include in proto\.h inside an \#ifdef PERL\_CORE?

The Perl\_pp\_\* functions aren't exported \(as part of the API\)

I suspect I was doing this to be consistent with everything else\, which is
only declared if it's exported\.

By inspection\, the module isn't going to work on platforms where the exporting
is relevant \- in particular\, Win32\.

Replacing Perl\_pp\_return with PL\_ppaddr\[OP\_RETURN\] would solve both\.

Nicholas Clark

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @cpansprout

I’m marking this as rejected\, since it is a CPAN\, rather than a perl\,
problem\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'rejected'
